src/System.CommandLine/Binding/SpecificSymbolValueSource.cs

```diff
@@ -1,34 +1,47 @@
 ﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Linq;
+
 namespace System.CommandLine.Binding
 {
-    internal class OptionValueSource : IValueSource
+    internal class SpecificSymbolValueSource : IValueSource
     {
-        public OptionValueSource(IOption option)
+        public SpecificSymbolValueSource(IValueDescriptor valueDescriptor)
         {
-            Option = option;
+            ValueDescriptor = valueDescriptor;
         }
 
-        public IOption Option { get; }
+        public IValueDescriptor ValueDescriptor { get; }
 
-        public bool TryGetValue(
-            IValueDescriptor valueDescriptor, 
-            BindingContext bindingContext, 
+        public bool TryGetValue(IValueDescriptor valueDescriptor,
+            BindingContext bindingContext,
             out object boundValue)
         {
-            var result = bindingContext.ParseResult.FindResultFor(Option);
-
-            switch (result)
+            valueDescriptor = ValueDescriptor;
```

Comment:
This always overwrites the passed in parameter with the property. Did you mean `valueDescriptor ??= ValueDescriptor;`?

Reply:
No, But I see that the code is misleading. I updated the implementation and during that also simplified it 😃